tokio/src/loom/mocked.rs

```diff
@@ -1,5 +1,32 @@
 pub(crate) use loom::*;
 
+pub(crate) mod sync {
+
+    pub(crate) use loom::sync::MutexGuard;
+
+    #[derive(Debug)]
+    pub(crate) struct Mutex<T>(loom::sync::Mutex<T>);
+
+    #[allow(dead_code)]
+    impl<T> Mutex<T> {
+        #[inline]
+        pub(crate) fn new(t: T) -> Mutex<T> {
+            Mutex(loom::sync::Mutex::new(t))
+        }
+
+        #[inline]
+        pub(crate) fn lock(&self) -> MutexGuard<'_, T> {
+            self.0.lock().unwrap()
+        }
```

Comment:
This doesn't really seem correct? Shouldn't it use `into_inner`? Or maybe it's good to panic on poisoning in loom tests?

Reply:
Poisoning in loom is good because we should never hit it.